### PR TITLE
return json in userpage() and fix userpage test

### PR DESCRIPTION
**Pull request by akrakman**

### CODE REVIEW

File: src/backend/app.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def userpage():
             return result, 400
         return result, 201
     user = page.get_user(username)  # request.method == "GET"
-    return user, username, 201
+    return json.dumps(user.__dict__), username, 201
```
[the remark this answers is not in the document]
Comment (akrakman) [reply src/backend/app.py:80]: Oh yeah forgot about that

[the remark this answers is not in the document]
Comment (akrakman) [reply src/backend/app.py:80]: But the User is not a list. How do you convert it?

File: src/backend/pages/login.py [hunk 2]
```diff
@@ -62,13 +61,22 @@ def logout(self) -> None:
         """Logout"""
 
 
+def validate_password(password: str) -> bool:
```
Comment (MinhPhan8803) [inline src/backend/pages/login.py:64]: The validate functions could be in their own separate files.

File: src/backend/tests/test_userpage.py [hunk 3]
```diff
@@ -1,60 +1,137 @@
...
+    email = "newemail@gmail.com"
+    userpage = None
+    main_page_staging = MainPageStaging()
 
-    @use_test
-    def test_valid_phone(self) -> bool:
-        """Test update_phone returns True and db entry is the same"""
-        assert validate_phone(self.phone) is True
-        assert self.userpage.update_phone(self.phone) is True
+    @use_database
```
Comment (MinhPhan8803) [inline src/backend/tests/test_userpage.py:19]: does this one have a `@use_test`? Otherwise it will write to db_prod

Comment (akrakman) [reply src/backend/tests/test_userpage.py:19]: ok will switch to use test

Comment (MinhPhan8803) [reply src/backend/tests/test_userpage.py:19]: Also `@use_test` should go before `@use_database` like 
`
@use_test
@use_database
do_stuff()
`

> ok will switch to use test

Comment (akrakman) [reply src/backend/tests/test_userpage.py:19]: I switch to use_test and everything fails now

**Review by MinhPhan8803 — APPROVED** [review]

**Pull request merged**